encoding/m3tsz/encoder.go

```diff
@@ -480,26 +480,20 @@ func (enc *encoder) trackNewSig(numSig uint8) uint8 {
 }
 
 func (enc *encoder) newBuffer(capacity int) []byte {
+	if capacity < 1 {
+		return nil
+	}
 	if bytesPool := enc.opts.BytesPool(); bytesPool != nil {
 		return bytesPool.Get(capacity)
 	}
 	return make([]byte, 0, capacity)
 }
 
 func (enc *encoder) Reset(start time.Time, capacity int) {
-	enc.ResetSetData(start, ts.Segment{Head: enc.newBuffer(capacity)})
+	enc.reset(start, enc.newBuffer(capacity))
 }
 
-func (enc *encoder) ResetSetData(start time.Time, data ts.Segment) ts.Segment {
-	// Return what this encoder owned by reference
-	segment := enc.segment(byRefResultType)
-
-	bytes := data.Head
-	if data.Tail != nil {
-		// NB(r): If you pass a tail this will cause a copy if
-		// head cannot fit the tail
-		bytes = append(bytes, data.Tail...)
-	}
+func (enc *encoder) reset(start time.Time, bytes []byte) {
 	enc.os.Reset(bytes)
```

Comment:
Is it possible that we end up calling `Reset(someTime, someSize)` while encoder is already holding on to some buffer? If yes, then perhaps we should add something like `if rb, _ := enc.os.Rawbytes(); rb != nil { /* return to pool */ }`.